### PR TITLE
Fix test for absent return values in thrift scheme

### PR DESCRIPTION
**Pull request by abhinav**
thriftrw now raises TypeError if return value is absent for non-void methods.

@breerly @blampe @junchaowu

### CODE REVIEW

**Comment by junchaowu** [comment]
lgtm thx for thefix

File: tests/schemes/test_thrift.py [hunk 1]
```diff
@@ -1031,7 +1031,13 @@ def testString(request):
         # values.
         exc = UnexpectedError
     else:
-        exc = ValueExpectedError
+        # If server is using thrift, it will be able to return an invalid
+        # response. thriftrw will fail with a TypeError on invalid values. For
+        # thrift, we'll check manually and raise ValueExpectedError.
+        if use_thriftrw_client:
+            exc = TypeError
```
Comment (blampe) [inline tests/schemes/test_thrift.py]: Should this be caught and re-raised as ValueExpectedError for consistency?

Comment (abhinav) [inline tests/schemes/test_thrift.py]: TypeError can also be raised if required arguments in a struct were absent, or on type mismatch. There's no error specific to "return value was absent" raised by thriftrw yet. I can fix that in the next version along with the unexpected exceptions support. Created uber/thriftrw-python#49.

Meanwhile this fix should suffice because this is a very specific corner case that will happen only if the server is using Apache Thrift, the client is using thriftrw, and the server forgets to return a value (which would normally be caught by their tests/manual checks). And when it does occur, there's no good way for the client to handle it except blow up.

Comment (blampe) [inline tests/schemes/test_thrift.py]: Sure sounds good!
